### ISSUE TITLE
[Feature Request] Advanced format strings

### ISSUE THREAD

**Issue by ImportTaste** [TumblThreeApp/TumblThree#173]
In gallery-dl, you can set a filename string like this: `{date:%Y-%m-%d %H'%M'%S}{title!t:? [/]/R  / /} {filename:Rtumblr_//R_/./}.{extension}`

Which means...

`{date`
`:`
`%Y-%m-%d %H'%M'%S` = YEAR _dash_ MONTH _dash_ DAY _‌ ‌ space ‌ ‌_ HOUR _apostrophe_ MINUTE _apostrophe_ SECONDS
`}`

`{title`
`!t` = trim the end result
`:`
`? [/]/` = prepend with "_space_ [" and append with "]"
`R  / /` = replace double spaces with a single space
`}`

`{filename`
`:`
`Rtumblr_//` = replace "tumblr_" with nothing
`R_/./` = replace "_" with "."
`}`

The end result is a filename like this: `2021-01-31 13'57'09 [Post Title] asdf123qwe456rty789.1280.png`
Or, if there is no post title: `2021-01-31 13'57'09 asdf123qwe456rty789.1280.png`

For TumblThree, my suggestion is a similar format like this: `%{d*Y-m-d H'M'S}%{p!t:? [/]/R  / /} %{f:Rtumblr_//R_/./}`

In other words, the variables for advanced formatting would be `%{variableletter}` instead of `%variableletter`

**Comment by thomas694** [comment]
Thanks for your feature request. We'll think about it.

Taking a first short look on it I can see that the whole set of options is python centric. So we could only mimic similar behavior.

Since we haven't received too much feedback yet, we can only assume that the already existing file rename functionality isn't used by too many users.
It would be interesting to know how you use our current file rename functionality. What patterns are you using and what option/feature are you missing most for your use case?

@ all
Do you use the file rename functionality and which patterns? What options are you missing?

**Comment by Hrxn** [comment]
Tumblr is not really a site that provides any meaningful metadata... so I don't get the point of this feature request, to be honest.

**Comment by thomas694** [comment]
The issue has been closed. You can still comment. Feel free to ask for reopening the issue if needed.
